bin/oe/rpc.rs

```diff
@@ -189,76 +176,116 @@ pub fn new_ws<D: rpc_apis::Dependencies>(
         }
         false => None,
     };
-    let start_result = rpc_servers::start_ws(
-        &addr,
-        handler,
-        allowed_origins,
-        allowed_hosts,
-        conf.max_connections,
-        rpc::WsExtractor::new(path.clone()),
-        rpc::WsExtractor::new(path.clone()),
-        rpc::WsStats::new(deps.stats.clone()),
-        conf.max_payload,
-    );
-
-    //	match start_result {
-    //		Ok(server) => Ok(Some(server)),
-    //		Err(rpc::ws::Error::Io(rpc::ws::ErrorKind::Io(ref err), _)) if err.kind() == io::ErrorKind::AddrInUse => Err(
-    //			format!("WebSockets address {} is already in use, make sure that another instance of an Ethereum client is not running or change the address using the --ws-port and --ws-interface options.", url)
-    //		),
-    //		Err(e) => Err(format!("WebSockets error: {:?}", e)),
-    //	}
-    match start_result {
-		Ok(server) => Ok(Some(server)),
-		Err(rpc::ws::Error::WsError(ws::Error {
-			                            kind: ws::ErrorKind::Io(ref err), ..
-		                            })) if err.kind() == io::ErrorKind::AddrInUse => Err(
-			format!("WebSockets address {} is already in use, make sure that another instance of an Ethereum client is not running or change the address using the --ws-port and --ws-interface options.", url)
-		),
-		Err(e) => Err(format!("WebSockets error: {:?}", e)),
-	}
+
+    let mut endpoints = conf.additional_endpoints;
+    endpoints.push(Endpoint {
+        interface: conf.interface,
+        port: conf.port,
+        apis: conf.apis,
+    });
+
+    let max_connections = conf.max_connections;
```

Comment:
need to unpack the struct here. It works without unpacking in 2021 edition, but doesn't in 2018, see https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a4adfd6d0ac4bf09519613391e7a1320